### PR TITLE
Implement NFTs V2 Arc endpoint

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1594

- [x] **revert graphql config changes before merge**

## What changed (plus any additional context for devs)
The major part of this PR includes consuming a new and faster nftV2 endpoint from Arc that @welps wrote. This cuts down nft fetching time by about 3x on average. There are still improvements to be made to it, but at least with his endpoint in, accounts with large amounts of NFTs should be able to see their collectibles again.

Other minor changes included removing the check or if the NFT section was empty, instead we now show an empty state and allow the user to peek through to the featured mint section if they wish to collect.

We needed to remove the check to remove the NFT section on empty because the NFT endpoint now fetches and sorts from the backend, so each time the user changed the sort method the Collectibles header would disappear.

We also now have a skeleton loader for when the NFT fetch is in-flight and no data is present. This helps let the user know that their collectibles are loading instead of just removing the section altogether. 

## Screen recordings / screenshots

Screenshot 2024-07-31 at 5 27 45 PM

https://github.com/user-attachments/assets/58f6e5a5-688a-48d1-bf95-7151a44c1463

### CODE REVIEW

**Review by benisgold — COMMENTED** [review]
looks pretty good, just minor comments. i saw a few places where we're relying on react query's `isFetching`- just want to double check that that's what we want vs. `isLoading`. also i noticed in your screenshot that the height of the "Collect Now" button is messed up

File: src/components/asset-list/RecyclerAssetList2/NFTLoadingSkeleton.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            {
+              backgroundColor: opacity(labelTertiary, 0.04),
+            },
+          ]}
+        />
+        <View style={sx.textContainer}>
+          <View
+            style={[
+              sx.title,
+              { backgroundColor: opacity(labelTertiary, 0.08), width: deviceUtils.dimensions.width / getRandomBetween(2, 4) },
```
[the remark this answers is not in the document]
Comment (walmat) [reply src/components/asset-list/RecyclerAssetList2/NFTLoadingSkeleton.tsx:40]: adds a little flare for NFT collection name length.

**Comment by walmat** [comment]
> looks pretty good, just minor comments. i saw a few places where we're relying on react query's `isFetching`- just want to double check that that's what we want vs. `isLoading`. also i noticed in your screenshot that the height of the "Collect Now" button is messed up

Yeah I played around with this a bit and think isLoading would remove the data on pull to referesh whereas isFetching wouldn't. I think that's the behavior we want.

**Review by benisgold — COMMENTED** [review]
It takes a second or two for the context menu text to switch from one option to the other. It should switch immediately after the user makes the selection even if the data is still loading. not sure what's causing the lag

https://github.com/user-attachments/assets/101cb0eb-e068-40bb-ba74-1564801928d3

**Comment by walmat** [comment]
> It takes a second or two for the context menu text to switch from one option to the other. It should switch immediately after the user makes the selection even if the data is still loading. not sure what's causing the lag
> 
>  Simulator.Screen.Recording.-.iPhone.15.Pro.-.2024-08-09.at.16.49.06.mp4

this is fixed now

**Review by benisgold — APPROVED** [review]

**Pull request merged**